src/lib/picker/category.component.ts

```diff
@@ -126,12 +161,36 @@ export class CategoryComponent implements OnInit {
       // this.labelSpanStyles = { position: 'absolute' };
     }
   }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.emojis?.currentValue?.length !== changes.emojis?.previousValue?.length) {
+      this.ngAfterViewInit();
+    }
+  }
+
+  ngAfterViewInit() {
+    if (!this.virtualize || !this.emojis?.length) {
+      return;
+    }
+
+    const parent = this.container.nativeElement.parentNode.parentNode;
+    const { width } = parent.getBoundingClientRect();
+
+    const rows = Math.ceil((this.emojis.length * (this.emojiSize + 12)) / width);
+
+    this.containerStyles = {
+      ...this.containerStyles,
+      minHeight: `${rows * (this.emojiSize + 12) + 28}px`,
```

Comment:
this is leaving a gap here

![image](https://user-images.githubusercontent.com/1400464/111943140-080f2280-8a92-11eb-860a-ee64706cc055.png)

I think in practice its more like 8px padding because they overlap

Reply:
Oh, I didn't even notice that the emoji padding overlapped. I'll take a look

Reply:
its either the overlap or the perRow math should fix the extra padding 🤷

Reply:
All good now. The extra space came from the emoji component. When using virtualize it was calculating the height of all emojis which includes obsolete or hidden emojis. I made sure to filter those out before height calculations are made